### PR TITLE
cmd/operator-sdk/olmcatalog/gen-csv.go: --write-config to write CSV config to disk

### LINKED ISSUE #1334: unable to generate olm-csv if no CRD are defined
## Bug Report

**What did you do?**
I wanted to generate the OCL CSV. I have an operator that does not define any new CRD. I got the following output:
```
operator-sdk olm-catalog gen-csv --csv-version 0.1.0
INFO[0000] Generating CSV manifest version 0.1.0        
Error: catalog scaffold failed: (lstat deploy/crds: no such file or directory)
Usage:
  operator-sdk olm-catalog gen-csv [flags]

Flags:
      --csv-config string     Path to CSV config file. Defaults to deploy/olm-catalog/csv-config.yaml
      --csv-version string    Semantic version of the CSV
      --from-version string   Semantic version of an existing CSV to use as a base
  -h, --help                  help for gen-csv
      --update-crds           Update CRD manifests in deploy/{operator-name}/{csv-version} the using latest API's
``` 

**What did you expect to see?**
Not sure, this was the first time I was running this command, not an error though.

**What did you see instead? Under which circumstances?**
see the output above.

The general question is: can OLM be used with an operator that does not define any new CRD? If yes the `operato-sdk olm-catalog gen-cvs` command should work.

**Environment**
* operator-sdk version:
```
operator-sdk --version
operator-sdk version v0.6.0+git
```
* go version:

```
go version
go version go1.11.6 linux/amd64
```

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:**
`internal/pkg/scaffold/olm-catalog`: Use a `CSVConfigFile` in a CSV scaffold directly instead of reading from a file. `CSVConfig` is now a scaffold that contains a `CSVConfigFile` with the actual data to write.

`cmd/operator-sdk/olmcatalog/gen-csv.go`: Add `--write-csv-config` flag to both `olm-catalog` and `olm-catalog gen-csv` so users can write a CSV config to a default path or the one passed to `--write-csv-config`.

**Motivation for the change:** #1259 does not directly pertain to the changes here but brought to my attention that CSV config files are not well understood. `--write-config` should help because it creates a default config with all fields populated.

### CODE REVIEW

**Comment by estroz** [comment]
After thinking about this further, its probably better to have a `olm-catalog write-csv-config` command so a user doesn't have to (re)write their CSV just to get a config.

**Comment by estroz** [comment]
/hold

Going to wait on this until we figure out configuration file issues.

File: cmd/operator-sdk/olmcatalog/gen-csv.go [hunk 1]
```diff
@@ -56,18 +57,18 @@ Configure CSV generation by writing a config file 'deploy/olm-catalog/csv-config
 	genCSVCmd.Flags().StringVar(&csvVersion, "csv-version", "", "Semantic version of the CSV")
 	genCSVCmd.MarkFlagRequired("csv-version")
 	genCSVCmd.Flags().StringVar(&fromVersion, "from-version", "", "Semantic version of an existing CSV to use as a base")
-	genCSVCmd.Flags().StringVar(&csvConfigPath, "csv-config", "", "Path to CSV config file. Defaults to deploy/olm-catalog/csv-config.yaml")
+	genCSVCmd.Flags().StringVar(&csvConfigPath, "csv-config", "", "Path to CSV config file. If unset, default file paths are used")
```
Comment (camilamacedo86) [inline cmd/operator-sdk/olmcatalog/gen-csv.go]: I think is good to keep the default path in the info.

File: cmd/operator-sdk/olmcatalog/cmd.go [hunk 2]
```diff
@@ -15,16 +15,70 @@
...
+	catalog "github.com/operator-framework/operator-sdk/internal/pkg/scaffold/olm-catalog"
+	"github.com/operator-framework/operator-sdk/internal/util/projutil"
+
+	log "github.com/sirupsen/logrus"
 	"github.com/spf13/cobra"
 )
 
+var (
+	writeCSVConfigOpt = "write-csv-config"
+
```
Comment (camilamacedo86) [inline cmd/operator-sdk/olmcatalog/cmd.go]: space

File: cmd/operator-sdk/olmcatalog/cmd.go [hunk 3]
```diff
@@ -15,16 +15,70 @@
...
 		Long: `The operator-sdk olm-catalog command invokes a command to perform
 Catalog related actions.`,
+		RunE: olmCatalogFunc,
 	}
 	cmd.AddCommand(newGenCSVCmd())
+
+	i, err := (&catalog.CSVConfig{}).GetInput()
+	if err != nil {
+		log.Fatalf("Error retrieving CSV config path: %v", err)
+	}
```
Comment (camilamacedo86) [inline cmd/operator-sdk/olmcatalog/cmd.go]: Could not the path be an attribute of the struct and the check be made in the Validate func of it?

File: cmd/operator-sdk/olmcatalog/cmd.go [hunk 4]
```diff
@@ -15,16 +15,70 @@
...
+		}
+		if projutil.IsOperatorGo() {
+			cfg.Repo = projutil.CheckAndGetProjectGoPkg()
+		}
+		s := &scaffold.Scaffold{}
+		if err := writeConfig(s, cfg); err != nil {
+			return err
+		}
+	}
+	return nil
```
Comment (camilamacedo86) [inline cmd/operator-sdk/olmcatalog/cmd.go]: Is it creating a new input.Config? Could not it be a NewConfig in its struc, or at least has a more meaningful name?

File: cmd/operator-sdk/olmcatalog/cmd.go [hunk 5]
```diff
@@ -15,16 +15,70 @@
...
+}
+func writeConfig(s *scaffold.Scaffold, cfg *input.Config) error {
```
Comment (camilamacedo86) [inline cmd/operator-sdk/olmcatalog/cmd.go]:
```suggestion
func writeCSVConfig(s *scaffold.Scaffold, cfg *input.Config) error {
```

File: cmd/operator-sdk/olmcatalog/gen-csv.go [hunk 6]
```diff
@@ -79,30 +80,46 @@ func genCSVFunc(cmd *cobra.Command, args []string) error {
...
+		csvCfgFile, err = catalog.GetCSVConfigFile(csvConfigPath)
+		if err != nil {
+			return err
+		}
+	} else if err != nil && csvConfigPath != "" {
+		return fmt.Errorf("error reading CSV config %s: %v", csvConfigPath, err)
+	} else {
+		log.Info("Using default CSV config.")
+		csvCfgFile = &catalog.CSVConfigFile{}
+	}
```
Comment (camilamacedo86) [inline cmd/operator-sdk/olmcatalog/gen-csv.go]: could not it be improved? 
By default, the default CSVFile be set then if do not find it or the customized one return an error? 
Could not the logic be when the new struct is created?

**Pull request closed without merging**